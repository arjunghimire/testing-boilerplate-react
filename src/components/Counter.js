import React, { Component } from "react";

class Counter extends Component {
  state = {
    counter: 0
  };
  handleIncDec = amount => () =>
    this.setState(prevState => ({
      counter: prevState.counter + amount
    }));
  handleIncrement = this.handleIncDec(1);
  handleDecrement = this.handleIncDec(-1);

  render() {
    const { counter } = this.state;

    return (
      <div className="count">
        <p className="counter">Count: {counter}</p>
        <button className="increment" onClick={this.handleIncrement}>
          Increment
        </button>
        <button className="decrement" onClick={this.handleDecrement}>
          Decrement
        </button>
      </div>
    );
  }
}

export default Counter;
