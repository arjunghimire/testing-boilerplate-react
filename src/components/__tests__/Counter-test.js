import React from "react";
import { shallow } from "enzyme";
import Counter from "../Counter";
import renderer from "react-test-renderer";

describe("Counter Component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render without throwing an error", () => {
    const wrapper = shallow(<Counter />);
    const countState = wrapper.state().counter;
    expect(countState).toEqual(0);
  });

  it("should respond to change event and increase value by 1", () => {
    const wrapper = shallow(<Counter />);
    wrapper.find("button.increment").simulate("click");
    const text = wrapper.find(".counter").text();
    expect(text).toEqual("Count: 1");
  });
  it("should respond to change event and decrease value by 1", () => {
    const wrapper = shallow(<Counter />);
    wrapper.find("button.decrement").simulate("click");
    const text = wrapper.find(".counter").text();
    expect(text).toEqual("Count: -1");
  });
});
