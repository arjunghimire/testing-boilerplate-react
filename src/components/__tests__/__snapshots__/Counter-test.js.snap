// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component renders correctly 1`] = `
<div
  className="count"
>
  <p
    className="counter"
  >
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
